src/easy.rs

```diff
@@ -2894,6 +2945,50 @@ extern fn easy_progress_cb(data: *mut c_void,
     })
 }
 
+extern fn easy_ssl_ctx_cb(handle: *mut curl_sys::CURL,
+                          ssl_ctx: *mut c_void,
+                          data: *mut c_void) -> curl_sys::CURLcode {
+
+    ssl_ctx_cb(handle, ssl_ctx, data, |ssl_ctx| unsafe {
+        match (*(data as *mut EasyData)).ssl_ctx.as_mut() {
+            Some(f) => f(ssl_ctx),
+            // If the callback isn't set we just tell CURL to
+            // continue.
+            None => Ok(()),
+        }
+    })
+}
+
+// TODO: same thing as `debug_cb`: can we expose `handle`?
+fn ssl_ctx_cb<F>(_handle: *mut curl_sys::CURL,
+                 ssl_ctx: *mut c_void,
+                 data: *mut c_void,
+                 f: F) -> curl_sys::CURLcode
+    where F: FnOnce(*mut c_void) -> Result<(), Error>
+{
+    if data.is_null() {
+        return curl_sys::CURLE_OK;
+    }
+
+    let result = panic::catch(|| {
+        f(ssl_ctx)
+    });
+
+    // Default to a generic SSL error in case of panic. This shouldn't
+    // really matter since the error should be propagated later on but
+    // better safe than sorry...
+    let result =
+        match result {
+            Some(r) => r,
+            None => Err(Error::new(curl_sys::CURLE_SSL_CONNECT_ERROR)),
+        };
+
+    match result {
+        Ok(()) => curl_sys::CURLE_OK,
+        Err(e) => e.code(),
+    }
```

Comment:
Perhaps this could just be:

```rust
match result {
    Some(Ok(())) => curl_sys::CURLE_OK,
    Some(Err(e)) => e.code(),
    None => curl_sys::CURLE_SYS_CONNECT_ERROR,
}
```